### ISSUE TITLE
False positives for hyphenated function names in function-calc-no-unspaced-operator

### ISSUE THREAD

**Issue by ilyavolodin** [stylelint/stylelint#1878]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where `function-calc-no-unspaced-operator` is incorrectly triggers on valid `calc` call.

> Which rule, if any, is this issue related to?

`function-calc-no-unspaced-operator`

> What CSS is needed to reproduce this issue?

```css
h1 {
  left: calc(100% - #{map-get($container-max-widths, xl)});
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "extends": "stylelint-config-standard"
}
```

> Which version of stylelint are you using?

`7.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --syntax scss *.css`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS maps.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
_hero.scss
 30:28  ✖  Expected single space before "-" operator   function-calc-no-unspaced-operator
 30:28  ✖  Expected single space after "-" operator    function-calc-no-unspaced-operator
```

**Comment by evilebottnawi** [comment]
@ilyavolodin Thanks for reporting, confirm bug.

**Comment by jeddy3** [comment]
@ilyavolodin Feel free to [contribute](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-bugs) a fix for this.

**Comment by evilebottnawi** [comment]
Done in https://github.com/stylelint/stylelint/pull/1984
